modules/javafx.web/src/main/native/Source/WebCore/page/DOMWindow.cpp

```diff
@@ -848,22 +848,18 @@ ExceptionOr<Storage*> DOMWindow::localStorage()
         return Exception { SecurityError };
 
     auto* page = document->page();
-    // FIXME: We should consider supporting access/modification to local storage
-    // after calling window.close(). See <https://bugs.webkit.org/show_bug.cgi?id=135330>.
-    if (!page || !page->isClosing()) {
-        if (m_localStorage)
-            return m_localStorage.get();
-    }
-
     if (!page)
         return nullptr;
 
-    if (page->isClosing())
-        return nullptr;
-
+    // Check if localstorage setting is disabled, then return nullptr
     if (!page->settings().localStorageEnabled())
         return nullptr;
 
+    // FIXME: We should consider supporting access/modification to local storage
+    // after calling window.close(). See <https://bugs.webkit.org/show_bug.cgi?id=135330>.
+    if (m_localStorage)
+            return m_localStorage.get();
```

Comment:
This is not needed here if you take my suggestion of doing it above.

Reply:
Kindly look code 👍 
ExceptionOr<Storage*> DOMWindow::localStorage()
{
    if (!isCurrentlyDisplayedInFrame())
        return nullptr;

    RefPtr document = this->document();
    if (!document)
        return nullptr;

    if (!document->securityOrigin().canAccessLocalStorage(nullptr))
        return Exception { SecurityError };

    // FIXME: We should consider supporting access/modification to local storage
    // after calling window.close(). See <https://bugs.webkit.org/show_bug.cgi?id=135330>.
    if (m_localStorage)
        return m_localStorage.get();

    auto* page = document->page();
    if (!page)
        return nullptr;

    // Check if localstorage setting is disabled, then return nullptr
    if (!page->settings().localStorageEnabled())
        return nullptr;

    auto storageArea = page->storageNamespaceProvider().localStorageArea(*document);
    m_localStorage = Storage::create(*this, WTFMove(storageArea));
    return m_localStorage.get();
}
 is it now ok?